torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -6,16 +6,134 @@ namespace rpc {
 
 constexpr size_t WorkerInfo::MAX_NAME_LEN;
 
+// TODO(osalpekar): Tune RetryDuration to be something in seconds per
+// team recommendations. Kept at 100ms for the sake of Milestone 1.
+const auto kRpcRetryDuration = std::chrono::milliseconds(100);
+// kBackoff is the multiplier for the exponential backoff algorithm.
+const float kBackoff = 1.2;
+
 RpcAgent::RpcAgent(
     WorkerInfo workerId,
     std::unique_ptr<RequestCallback> cb,
     std::chrono::milliseconds rpcTimeout)
     : workerInfo_(std::move(workerId)),
       cb_(std::move(cb)),
       rpcTimeout_(rpcTimeout),
-      profilingEnabled_(false) {}
+      profilingEnabled_(false) {
+  {
+    std::lock_guard<std::mutex> lock(rpcRetryMutex_);
+    rpcAgentRunning_.store(true);
+  }
+  rpcRetryThread_ = std::thread(&RpcAgent::retryExpiredRpcs, this);
+}
+
+RpcAgent::~RpcAgent() {
+  {
```

Comment:
This seems like it potentially belongs in the equivalent of shutdown() (I realize this is virtual = 0 now).
It's not awesome practice to block in the destructor like this.

It's probably ok as-is now, but probably want a future change where the subclasses call this super classes's shutdown(), and move to there.